neuralpredictors/layers/activations.py

```diff
@@ -186,3 +186,12 @@ def forward(self, *args, data_key=None, **kwargs):
 
     def regularizer(self, data_key):
         return self[data_key].smoothness_regularizer()
+
+
+class SoftThreshold(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.zero = torch.nn.parameter.Parameter(torch.tensor(0, dtype=torch.float32), requires_grad=False)
```

Comment:
`Parameter` is in `torch.nn.parameter` (https://pytorch.org/docs/stable/generated/torch.nn.parameter.Parameter.html#torch.nn.parameter.Parameter).

Yes, always gradientless (constant). That is a great suggestion, I revised the code accordingly.

Reply:
> Re `independent_bn_bias`: Do you mean that setting it to `True` or `False` has no effect on the model? Both from the code and documentation it seems that this argument would additionally define a bias (and scale) if it is set to `False`. So I am not sure if the argument has no effect on the model.
> 
> Or do you think it does not make sense to have it? To me, it is kind of weird that when `independent_bn_bias=False`, which supposedly means that bias and bn are NOT independent, there is suddenly a bias defined independently in addition to the batchnorm. But I am not familiar with the model and the code at all, so I might be missing something.
> 
> For the future fix/improvement, would you suggest removing the argument and directly using [this code block](https://github.com/MaxFBurg/neuralpredictors/blob/e29b7997be1188de3674cb9455c4df12ea974914/neuralpredictors/layers/cores/conv2d.py#L212-L223) or you had other ideas?

I think it does not make much sense to have this argument, so I would suggest exactly what you described in the last paragraph above. Having `independent_bn_bias` as an argument is superficial, and in some sense artificial/technical: what you are interested in is configuring batchnorm such that it can have any combination of scale=True/False and bias=True/False. In the end, it does not matter if this is done by `nn.BatchNorm2d` or by turning both off in `nn.BatchNorm2d` and then adding `Bias2DLayer` and `Scale2DLayer` (or any combination). So from the perspective of the users: they do not care how this is done internally.

If you look at the code block you cited above: this handles all combinations of bias and scale, so this code block is sufficient to configure batchnorm behavior in any way you like. (It just looks this complicated because in `nn.BatchNorm2d` you can just turn on/off both scale and bias at the same time, so if you just want to have scale xor bias, you have to turn both off and introduce a scale / bias layer yourself. In tensorflow, you could configure scale and bias in batchnorm independently). 

So bottom line: remove `independent_bn_bias`, because it is superficial / does not make sense to have it, as all cases / configurations are handled in [this code block](https://github.com/MaxFBurg/neuralpredictors/blob/e29b7997be1188de3674cb9455c4df12ea974914/neuralpredictors/layers/cores/conv2d.py#L212-L223) .